File: versioned_docs/version-1.21.x/examples/cpp/src/lifecycle_policy_mode.cc
#include <cassert>
#include <reduct/client.h>

using reduct::Error;
using reduct::IClient;

int main() {
  auto client = IClient::Build("http://127.0.0.1:8383", {.api_token = "my-token"});

  auto err = client->SetLifecycleMode("my-lifecycle", IClient::LifecycleMode::kDryRun);
  assert(err == Error::kOk);

  err = client->SetLifecycleMode("my-lifecycle", IClient::LifecycleMode::kDisabled);
  assert(err == Error::kOk);

  err = client->SetLifecycleMode("my-lifecycle", IClient::LifecycleMode::kEnabled);
  assert(err == Error::kOk);
}

File: versioned_docs/version-1.21.x/examples/go/src/lifecycle_policy_mode.go
package main

import (
	"context"

	reduct "github.com/reductstore/reduct-go"
	model "github.com/reductstore/reduct-go/model"
)

func main() {
	client := reduct.NewClient("http://localhost:8383", reduct.ClientOptions{APIToken: "my-token"})

	if err := client.SetLifecycleMode(context.Background(), "my-lifecycle", model.LifecycleModeDryRun); err != nil {
		panic(err)
	}
	if err := client.SetLifecycleMode(context.Background(), "my-lifecycle", model.LifecycleModeDisabled); err != nil {
		panic(err)
	}
	if err := client.SetLifecycleMode(context.Background(), "my-lifecycle", model.LifecycleModeEnabled); err != nil {
		panic(err)
	}
}
